src/common/TrackedOp.h

```diff
@@ -166,13 +170,15 @@ class TrackedOp {
     lock("TrackedOp::lock"),
     seq(0),
     warn_interval_multiplier(1),
-    is_tracked(false)
+    is_tracked(false),
+    is_registered(false)
   {
     RWLock::RLocker l(tracker->lock);
     if (tracker->tracking_enabled) {
       tracker->register_inflight_op(&xitem);
       events.push_back(make_pair(initiated_at, "initiated"));
-      is_tracked = true;
+      // Tells tracking_start() whether to set is_tracked or not
+      is_registered = true;
     }
```

Comment:
Hmm, this does solve the problem, but I think the better solution would be to move this code into a TrackedOp::register method and call it in OpTracker::create_request.  That way, we never have a half-constructed object registered in the first place and can avoid having both is_tracked and is_registered.